### ISSUE TITLE
Load/save workspace seems to be broken in v5.3.1

### ISSUE THREAD

**Issue by james-d-mitchell** [semigroups/Semigroups#960]
Doing

~~~
Test("tst/workspaces/save-workspace.tst");
quit;
gap -L tst/workspaces/test-output.w
~~~
causes:
~~~
❯ gap -L tst/workspaces/test-output.w
Stack trace (most recent call last):
#7    Object "gap", at 0x10440c277, in main + 1071
#6    Object "gap", at 0x10445af5f, in realmain + 79
#5    Object "gap", at 0x10445b1f3, in InitializeGap + 143
#4    Object "gap", at 0x1044f533f, in LoadWorkspace + 1707
#3    Object "gap", at 0x104418ab3, in LoadFunction + 199
#2    Object "libsystem_platform.dylib", at 0x1ac55ea23, in _sigtramp + 55
#1    Object "libsemigroups.2.dylib", at 0x105701823, in backward::SignalHandling::sig_handler(int, __siginfo*, void*) + 19
#0    Object "libsemigroups.2.dylib", at 0x10570187b, in backward::SignalHandling::handleSignal(int, __siginfo*, void*) + 59
[1]    50092 segmentation fault  /Users/jdm/gap/bin/gap.sh -m 1g -L tst/workspaces/test-output.w
~~~

This can also be seen here:

![Screenshot 2023-10-05 at 16 49 32](https://github.com/semigroups/Semigroups/assets/6686071/17288f4c-d4c8-4c2e-bb3f-697f09bf9716)

**Comment by james-d-mitchell** [comment]
Also present in v5.3.0

**Comment by james-d-mitchell** [comment]
Works in v5.2.1

**Comment by james-d-mitchell** [comment]
Git bisect yields:

~~~
6712254d97b0ff4e65a6e86caf3e81e72cab2008 is the first bad commit
commit 6712254d97b0ff4e65a6e86caf3e81e72cab2008
Author: James D. Mitchell <jdm3@st-andrews.ac.uk>
Date:   Tue Sep 12 13:23:05 2023 +0100

    kernel: fix mem leaks

 etc/tst-local-vars.py                     |  8 ++---
 etc/tst-unbind-local-vars.py              | 49 ++++++++++++++++++++++++++++++
 gapbind14/include/gapbind14/gapbind14.hpp | 11 ++++---
 gapbind14/include/gapbind14/to_cpp.hpp    | 12 +++++---
 gapbind14/src/gapbind14.cpp               | 34 +++++++++++++++++----
 tst/standard/attributes/homomorph.tst     | 50 +++++++++++++++++++++++++++++++
 tst/standard/semigroups/semifp.tst        | 24 +++++++++++----
 tst/standard/tools/display.tst            |  8 ++++-
 8 files changed, 171 insertions(+), 25 deletions(-)
 create mode 100755 etc/tst-unbind-local-vars.py
~~~

**Comment by james-d-mitchell** [comment]
Resolved by #962
